### ISSUE TITLE
Touch input

### ISSUE THREAD

**Issue by elmindreda** [glfw/glfw#42]

**Comment by nippoo** [comment]
Is this still slated for 3.2? :)

**Comment by elmindreda** [comment]
I hope so.

**Comment by elmindreda** [comment]
[Multitouch](http://www.x.org/wiki/Development/Documentation/Multitouch/)

**Comment by ec1oud** [comment]
Looks like not much has been happening with this lately, although on the touch branch it looked to be off to a good start (on Windows only?)  Having some second thoughts about the approach, or just too much work/not enough time, or because there is a solution provided by some other library?

**Comment by elmindreda** [comment]
@ec1oud For my part, not enough time and no hardware to test on.

**Comment by NeatSketch** [comment]
@elmindreda I have a touchscreen laptop with Windows 10 & Ubuntu installed. Is there any way I can help with testing?

**Comment by Renanse** [comment]
Interested in this as well.  Would love to be able to implement glfw-backed multi-touch input in Ardor3D.  Right now I'm only able to implement this in SWT land. :/

**Comment by torkeldanielsson** [comment]
I played around with this and did a rebase of the touch branch on top of the latest master:
https://github.com/torkeldanielsson/glfw/tree/touch

It isnt't so clean (not every commit builds, there is a fixup commit that fixes misses during the rebase). End state should be as intended though. ..and let me know if there is an interest in a cleaner rebase and I can fix it :)

Thought: from "WM_TOUCH is totally bananas" it seems like maybe WM_TOUCH is not the right choice? I would guess win8/win10 is a fine target for this. Or is it that WM_TOUCH is more similar to what is available on Linux?

**Comment by cypherdare** [comment]
I don't currently have a touch screen to test. But does the existing GLFW treat screen touches as a mouse click and drag, or is it ignored?

**Comment by rseymour** [comment]
I have a razer blade stealth 2018 w/ a QHD screen and Ubuntu Studio and I'm also up for testing.

**Comment by raysan5** [comment]
Any progress on multi-touch support for desktop platforms? Is it even possible coding in C?

**Comment by alectron** [comment]
Random thought - with VCV (a music / modular eurorack style software ) requiring this to be addressed on glfw officially to make use of multi-touch for VCV... and the VCV community being quite active and there being regular requests for touch support: 
Is there a capable person on the glfw project who might be keen to work to work & deliver this if there was payment availeable ?  (as to which we'd first need an estimate as to what that capable person would find reasonable to actually work on this and deliver this feature to glfw...  second would be to set this up as a indiegogo/kickstarter) - feel free to reach out with a number to me to post about the option to see if there's enough interest, or post the vcv community directly.
Chur.

**Comment by linkmauve** [comment]
As you can see in #532, back in 2015 there was an implementation for both Wayland and Windows, only missing X11 and Cocoa support. I think you may want to direct your search towards developers for these two platforms.

I can rebase the Wayland part to GLFW master whenever you want, but I don’t currently have any project using touch.

**Comment by alectron** [comment]
n Andrew's words:
https://github.com/VCVRack/Rack/issues/960#issuecomment-604954414
If anyone feels up for that, get in touch if the above random idea is interesting.
Chur.

**Comment by vuvukoma** [comment]
Hi,
Is there a devel branch for adding touch events handling for Wayland? 

Thanks.

**Comment by parkerlreed** [comment]
> I just implemented touch support for X11, integrated Windows support( #1736 ) and Wayland support ( #532 )
> 
> This is the rebase of these codes integrated together based on GLFW 3.4: https://github.com/xfangfang/glfw/tree/touch

Should this be a drop in replacement? I used your fork and branch to install glfw-git on Arch. Still observing no touch in Blender/SDR++.

**Comment by xfangfang** [comment]
> Should this be a drop in replacement?

No, There is a callback function to get the touch event: glfwSetTouchCallback

Maybe you can add the callback function and then simulate the mouse click in the callback.

**Comment by davidgiga1993** [comment]
What is the current status of this issue? Given that there is a implementation available, would this just needs to be merged?
